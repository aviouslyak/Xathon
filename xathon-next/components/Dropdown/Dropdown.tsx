import React from "react";

interface Props {
  returnedItems: string[];
  setQuery: React.Dispatch<React.SetStateAction<string>>;
}
const Dropdown: React.FC<Props> = ({ returnedItems, setQuery }) => {
  return (
    <div
      data-testid="dropdown-container"
      className="mt-0 flex flex-col items-start z-500 absolute w-1/2"
    >
      {returnedItems.slice(0, 5).map((item, i) => {
        return (
          <button
            key={i}
            className={`text-left ${
              i === 4 || i === returnedItems.length - 1 ? "rounded-b-lg" : ""
            } p-2 bg-pink-200 hover:bg-orange-400 shadow-lg shadow-pink-900 w-full overflow-ellipsis block whitespace-nowrap overflow-hidden`}
            onClick={() => {
              setQuery(item);
            }}
          >
            {item.startsWith("0x") ? `Contract with address '${item}'` : item}
          </button>
        );
      })}
    </div>
  );
};

export default Dropdown;
