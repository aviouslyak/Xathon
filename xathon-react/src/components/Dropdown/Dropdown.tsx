import React from "react";

interface Props {
  returnedItems: string[];
  setQuery: React.Dispatch<React.SetStateAction<string>>;
}
const Dropdown: React.FC<Props> = ({ returnedItems, setQuery }) => {
  return (
    <div
      data-testid="dropdown-container"
      className="mt-0 responsive-component-width flex flex-col items-start"
    >
      {returnedItems.slice(0, 5).map((item, i) => {
        return (
          <button
            key={i}
            className={`text-left ${
              i === 4 || i === returnedItems.length - 1 ? "rounded-b-lg" : ""
            } p-2 dark:hover:bg-gray-900 dark:bg-gray-700 bg-gray-200 hover:bg-gray-300 w-full overflow-ellipsis block whitespace-nowrap overflow-hidden`}
            onClick={() => {
              setQuery(item);
            }}
          >
            {item.startsWith("0x") ? `Contract with address '${item}'` : item}
          </button>
        );
      })}
    </div>
  );
};

export default Dropdown;
