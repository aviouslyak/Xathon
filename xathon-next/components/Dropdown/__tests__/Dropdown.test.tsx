import { fireEvent, render, screen } from "@testing-library/react";
import Dropdown from "../Dropdown";

const returnedItems: string[] = [];
for (let index = 0; index < 4; index++) {
  returnedItems.push(`Item ${index}`);
}
returnedItems.push("0x0000");
returnedItems.push("Will not find");

const setQueryMock = jest.fn();

describe("<Dropdown />", () => {
  let component: HTMLElement;
  let items: HTMLElement[];
  beforeEach(() => {
    render(<Dropdown returnedItems={returnedItems} setQuery={setQueryMock} />);
    component = screen.getByTestId("dropdown-container");
    items = screen.getAllByRole("button");
  });

  test("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  test("has five items", () => {
    expect(items.length).toBe(5);
  });

  test("does not have six items", () => {
    expect(items.length).not.toBe(6);
  });

  test("click on button fires setQuery", () => {
    const firstItem = items[0];
    fireEvent.click(firstItem);
    expect(setQueryMock).toHaveBeenCalledWith("Item 0");
  });

  test("displays contract text", () => {
    const contractItem = screen.getByText("Contract with address '0x0000'");
    expect(contractItem).toBeInTheDocument();
  });
  test("final item has correct class 'rounded-b-lg'", () => {
    const lastItem = items[items.length - 1];
    expect(lastItem).toHaveClass("rounded-b-lg");
  });

  test("other items does not have class 'rounded-b-lg'", () => {
    const middleItem = items[2];
    expect(middleItem).not.toHaveClass("rounded-b-lg");
  });
});
