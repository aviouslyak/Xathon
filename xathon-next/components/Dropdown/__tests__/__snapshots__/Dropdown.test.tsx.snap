// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> matches snapshot 1`] = `
<div
  class="mt-0 flex flex-col items-start z-500 absolute w-1/2"
  data-testid="dropdown-container"
>
  <button
    class="text-left  p-2 bg-pink-200 hover:bg-orange-400 shadow-lg shadow-pink-900 w-full overflow-ellipsis block whitespace-nowrap overflow-hidden"
  >
    Item 0
  </button>
  <button
    class="text-left  p-2 bg-pink-200 hover:bg-orange-400 shadow-lg shadow-pink-900 w-full overflow-ellipsis block whitespace-nowrap overflow-hidden"
  >
    Item 1
  </button>
  <button
    class="text-left  p-2 bg-pink-200 hover:bg-orange-400 shadow-lg shadow-pink-900 w-full overflow-ellipsis block whitespace-nowrap overflow-hidden"
  >
    Item 2
  </button>
  <button
    class="text-left  p-2 bg-pink-200 hover:bg-orange-400 shadow-lg shadow-pink-900 w-full overflow-ellipsis block whitespace-nowrap overflow-hidden"
  >
    Item 3
  </button>
  <button
    class="text-left rounded-b-lg p-2 bg-pink-200 hover:bg-orange-400 shadow-lg shadow-pink-900 w-full overflow-ellipsis block whitespace-nowrap overflow-hidden"
  >
    Contract with address '0x0000'
  </button>
</div>
`;
